src/attack.c

```diff
@@ -655,8 +662,21 @@ attack_host (struct scan_globals *globals, struct in6_addr *ip, GSList *vhosts,
           return;
         }
 
+      if (check_kb_inconsistency (main_kb) != 0)
+        {
+          // As long as we don't have a proper communication channel
+          // to our ancestors we just kill our parent and ourselves
+          // (but let our grandparents live).
+          // To prevent duplicate results we don't let ACT_END run.
+          kill (parent, SIGTERM);
+          // just in case
+          raise (SIGTERM);
+          return;
```

Comment:
```suggestion
          kill (parent, SIGUSR1);
          raise (SIGUSR1);
```
Killing the parent will be just ignored and killing itself disables the clean up. Currently running scripts won't get stopped. Instead sending a `SIGUSR1` to the parent and itself will work as a stop scan signal, which should work in this case, as the scheduler will call `scan_is_stopped()` directly after this block.

Reply:
An alternative to start the stop scan chain, would be to use the parent pid as group pid. Like in openvas.c, you can call directly 
```suggestion
          killpg (parent, SIGUSR1);
```